### PR TITLE
Learnable time constants

### PR DESCRIPTION
**Pull request by Huizerd**
Hi,

Very nice package you have made here! I'd like to help out, so I made a draft for learnable time constants. Please let me know what you think! (Solves #39)

### CODE REVIEW

**Comment by jeshraghian** [comment]
Hey @Huizerd, 

This is beautiful! I appreciate this a lot; especially taking the time to handle each neuron model. Everything looks great to me. 

I'm cool with leaving the beta/R/C in the `Lapicque` model unlearnable, as that is intended to just be a physical RC circuit. Users can just rely on `Leaky` if they want a learnable neuron there.

You've suggested perhaps passing the time constant to `torch.exp(-1 / self.beta.clamp_min(0))`. I've been thinking about this for a while. I've settled on just passing the direct decay rate as it feels more intuitive (to me, at least).

**Comment by Huizerd** [comment]
My pleasure! You're right about the decay rates being more intuitive. Another way to make learning smoother could be to compute the actual decay rate as `torch.sigmoid(alpha)`, such that it always stays in [0, 1], but then again you have the same problem of having to give an unintuitive `alpha` (say 0.0 for 0.5 decay rate). Just something to think about :)
